From bc6b3cf3f0c541d074de1614e8fffa906a604f3f Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Wed, 15 Aug 2018 17:57:45 +0800
Subject: [PATCH 4478/5242] MLK-19227-4: HDMI: edid function is not supported
 by i.MX8QM A0 chip

commit  3b871018ef18a99a4251f89b75ec6f4f58258d12 from
https://source.codeaurora.org/external/imx/linux-imx.git

EDID function is not supported by i.MX8QM A0 SOC chip.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit 6aeff3508919d584c6ce5661b14fadf3187298e7)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 6e02c12..8d8f62a 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -26,6 +26,8 @@
 #include "imx-dp.h"
 #include "../imx-drm.h"
 
+#define B0_SILICON_ID			0x11
+
 struct drm_display_mode *g_mode;
 
 static struct drm_display_mode edid_cea_modes[] = {
@@ -1247,6 +1249,10 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 
 	hdp->no_edid = of_property_read_bool(pdev->dev.of_node, "fsl,no_edid");
 
+	/* EDID function is not supported by iMX8QM A0 */
+	if (cpu_is_imx8qm() && (imx8_get_soc_revision() < B0_SILICON_ID))
+		hdp->no_edid = true;
+
 	ret = of_property_read_u32(pdev->dev.of_node,
 				       "lane_mapping",
 				       &hdp->lane_mapping);
-- 
1.7.9.5

